go-apps/meep-auth-svc/server/auth-svc.go

```diff
@@ -365,32 +367,58 @@ func cacheServicePermissions(cfg *PermissionsConfig) {
 	authSvc.cache.Services = make(map[string]map[string]*Permission)
 
 	for _, svc := range cfg.Services {
-		// Create new service + add it to service cache
-		svcMap := make(map[string]*Permission)
-		authSvc.cache.Services[svc.Name] = svcMap
+		// Get/Create service + add it to service cache
+		svcMap, found := authSvc.cache.Services[svc.Name]
+		if !found {
+			svcMap = make(map[string]*Permission)
+			authSvc.cache.Services[svc.Name] = svcMap
+		}
+
+		// Get API-specific prefix if present
+		apiPrefix := ""
+		if svc.Api != "" {
+			apiPrefix = svc.Api + "--"
+		}
 
 		// Service Endpoints
 		for _, ep := range svc.Endpoints {
-			// Cache service endpoint permissions
+			// Create service endpoint permissions
 			permission := new(Permission)
 			permission.Mode = ep.Mode
 			permission.Roles = make(map[string]string)
 			for role, access := range ep.Roles {
 				permission.Roles[role] = access
 			}
-			svcMap[ep.Name] = permission
 
-			// Add auth service route
-			route := new(AuthRoute)
-			route.Name = ep.Name
-			route.Prefix = false
-			route.Method = ep.Method
+			// Add auth service routes + cache service endpoint permissions
 			if svc.Sbox {
+				// Mep-specific sandbox service endpoint
+				route := new(AuthRoute)
+				route.Prefix = false
+				route.Method = ep.Method
+				route.Name = mepPrefix + apiPrefix + ep.Name
+				route.Pattern = "/{sbox}/{mep}" + svc.Path + ep.Path
+				routes = append(routes, route)
+				svcMap[route.Name] = permission
+
+				// Sandbox service endpoint
+				route = new(AuthRoute)
+				route.Prefix = false
+				route.Method = ep.Method
+				route.Name = apiPrefix + ep.Name
 				route.Pattern = "/{sbox}" + svc.Path + ep.Path
+				routes = append(routes, route)
+				svcMap[route.Name] = permission
 			} else {
+				// Global service endpoint
```

Comment:
lot of common code in the if/else... could be taken out of the if/else

Reply:
Agreed. Will try to address when working in this module again.